### PR TITLE
Run iOS simulator install as root

### PR DESCRIPTION
**Pull request by a-maurice**
The xcodes install can fail when not run as root, so do so.
***

***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: scripts/gha/test_simulator.py [hunk 1]
```diff
@@ -517,7 +517,7 @@ def _create_and_boot_simulator(apple_platform, device_name, device_os):
         print("::warning ::Unable to find %s version %s, will fall back to %s" % (apple_platform, device_os, available_versions[-1]))
       device_os = available_versions[-1]
 
-    args = ["xcodes", "runtimes", "install", "%s %s" % (apple_platform, device_os)]
+    args = ["sudo", "xcodes", "runtimes", "install", "%s %s" % (apple_platform, device_os)]
```
Comment (jonsimantov) [inline scripts/gha/test_simulator.py:520]: Update the info log message below to say it uses sudo; that way, if a user is running this on their machine, they won't be surprised by the password prompt.

Comment (a-maurice) [reply scripts/gha/test_simulator.py:520]: It prints out the args array, so the sudo should be included in the message

Comment (jonsimantov) [reply scripts/gha/test_simulator.py:520]: I guess that's fine, though not quite as user-friendly as it could be. :P

**Review by jonsimantov — APPROVED** [review]

**Comment by jonsimantov** [comment]
BTW the time this failed was when I ran "full" tests so you might need to do that...

**Pull request merged**